index.js

```diff
@@ -0,0 +1,319 @@
+  //. . 'Ord             ✅ * ',  // if ‘a’ satisfies Ord
+  //. . 'Semigroupoid    ❌   ',
+  //. . 'Category        ❌   ',
+  //. . 'Semigroup       ✅ * ',  // if ‘a’ satisfies Semigroup
+  //. . 'Monoid          ❌   ',
+  //. . 'Group           ❌   ',
+  //. . 'Filterable      ❌   ',
+  //. . 'Functor         ✅   ',
+  //. . 'Bifunctor       ✅   ',
+  //. . 'Profunctor      ❌   ',
+  //. . 'Apply           ✅ * ',  // if ‘a’ satisfies Semigroup
+  //. . 'Applicative     ✅ * ',  // if ‘a’ satisfies Monoid
+  //. . 'Chain           ❌   ',
+  //. . 'ChainRec        ❌   ',
+  //. . 'Monad           ❌   ',
+  //. . 'Alt             ❌   ',
+  //. . 'Plus            ❌   ',
+  //. . 'Alternative     ❌   ',
+  //. . 'Foldable        ✅   ',
+  //. . 'Traversable     ✅   ',
+  //. . 'Extend          ❌   ',
+  //. . 'Comonad         ❌   ',
+  //. . 'Contravariant   ❌   ' ]
+  //. ```
+
+  //# Constant :: TypeRep a -> a -> Constant a b
+  //.
+  //. Constant's sole data constructor. The [type representative][] makes
+  //. `Constant (M)` an [Applicative][]-compatible type representative if
+  //. `M` represents some monoidal type.
+  //.
+  //. ```javascript
+  //. > Constant (String) ('abc')
+  //. Constant (String) ('abc')
+  //.
+  //. > Constant (Number) (123)
+  //. Constant (Number) (123)
+  //. ```
+  const Constant = A => {
+    const prototype = {
+      /* eslint-disable key-spacing */
+      'constructor':            Constant$bound,
+      '@@type':                 constantTypeIdent,
+      '@@show':                 Constant$prototype$show,
+      'fantasy-land/map':       Constant$prototype$map,
+      'fantasy-land/bimap':     Constant$prototype$bimap,
+      'fantasy-land/reduce':    Constant$prototype$reduce,
+      'fantasy-land/traverse':  Constant$prototype$traverse,
+      /* eslint-enable key-spacing */
+    };
+
+    if (globalThis.process?.versions?.node != null) {
+      const inspect = Symbol.for ('nodejs.util.inspect.custom');
+      prototype[inspect] = Constant$prototype$show;
+    }
+
+    /* c8 ignore start */
+    if (typeof globalThis.Deno?.customInspect === 'symbol') {
+      const inspect = globalThis.Deno.customInspect;
+      prototype[inspect] = Constant$prototype$show;
+    }
+    /* c8 ignore stop */
+
+    function Constant$bound(value) {
+      const constant = Object.create (prototype);
+      if (Z.Setoid.test (value)) {
+        constant['fantasy-land/equals'] = Constant$prototype$equals;
+        if (Z.Ord.test (value)) {
+          constant['fantasy-land/lte'] = Constant$prototype$lte;
+        }
+      }
+      if (Z.Semigroup.test (value)) {
+        constant['fantasy-land/concat'] = Constant$prototype$concat;
+        constant['fantasy-land/ap'] = Constant$prototype$ap;
+      }
+      constant.value = value;
+      return constant;
+    }
+
+    //  XXX: sanctuary-show@3 only respects @@show on "objects". Fool it.
+    Constant$bound[Symbol.toStringTag] = 'Object';
```

Comment:
@Avaq, here's a workaround for the issue we discussed not so long ago.

Reply:
Would this become a standard way of working? It seems a bit hacky.

Reply:
sanctuary-js/sanctuary-show#28